tasks/src/main/scala/dagr/tasks/fgbio/GroupReadsByUmi.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks.fgbio
+
+import dagr.commons.CommonsDef._
+import dagr.core.tasksystem.Pipe
+import dagr.tasks.DataTypes.SamOrBam
+
+import scala.collection.mutable.ListBuffer
+
+/** Task to invoke GroupReadsByUmi. */
+class GroupReadsByUmi(val in:  PathToBam,
+                      val out: PathToBam,
+                      val rawTag: Option[String] = None,
+                      val assignTag: String,
+                      val minMapQ: Option[Int] = None,
+                      val strategy: String = "adjacency",
```

Comment:
Perhaps here it makes more sense to have an enum, similar to how we have `DownsamplingStrategy` in `DownsampleSam`.